File: array.js
// ARRAY CRUD
const students = ['Phu', 'Cai', 'Joyce'];

// Create
students.push("Guy");
console.log(students);

// Read
console.log(students[1]);

// Update
students[1] = "Super Cai";
console.log(students);

// Delete
// students.splice(1,1); // remove from index 1, 1 element
students.splice(1,2); // remove from index 1, 2 elements
console.log(students);

// Iterate
// JS arrow function: () => {}
// students.forEach(() => {});
students.forEach((student) => {
  console.log(`${student} is amazing!`);
});

const sentences = students.map((student) => {
  return `${student} is amazing!`;
});

console.log(sentences);

File: array.rb
# ARRAY CRUD
students = ['Phu', 'Cai', 'Joyce']

# Create
students.push("Guy")
students << "Mei"
p students

# Read
p students[1]

# Update
students[1] = "Super Cai"
p students

# Delete
students.delete_at(1)
# students.delete("Super Cai")
p students

# Iterate
students.each do |student|
  puts "#{student} is amazing!"
end

sentences = students.map do |student|
  "#{student} is amazing!"
end

p sentences